core/javax.cache/src/main/java/org/wso2/carbon/caching/impl/CacheImpl.java

```diff
@@ -831,7 +831,12 @@ public void expire(K key) {
                 log.warn("Exception occurred while expiring item from distributed cache. " + e.getMessage());
             }
         }
-        if (isIdle()) {
+        //keep the empty cache object if discardEmptyCachesProp is false
+        String discardEmptyCachesProp =
+                ServerConfiguration.getInstance().getFirstProperty(CachingConstants.DISCARD_EMPTY_CACHES);
+        boolean discardEmptyCaches =
+                (discardEmptyCachesProp == null) || Boolean.parseBoolean(discardEmptyCachesProp);
```

Comment:
we can use JavaUtils.isTrueExplicitly() directly

Reply:
since we need to check whether the property is null and if it is null we need to return true, we can keep this logic instead of using JavaUtils.isTrueExplicitly().